gsdk/src/signer/mod.rs

```diff
@@ -17,22 +17,55 @@
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 //! Gear api with signer
+
 use crate::{
     config::GearConfig,
     result::{Error, Result},
     Api,
 };
+use core::ops::Deref;
 pub use pair_signer::PairSigner;
 use sp_core::{crypto::Ss58Codec, sr25519::Pair, Pair as PairT};
 use sp_runtime::AccountId32;
+use std::sync::Arc;
 
 mod calls;
 mod pair_signer;
 mod rpc;
 mod utils;
 
+/// Signer representation that provides access to gear API.
+/// Implements low-level methods such as [`run_tx`](`SignerInner::run_tx`)
+/// and [`force_batch`](`SignerInner::force_batch`).
+/// Other higher-level calls are provided by [`Signer::sudo`],
+/// [`Signer::balance`], [`Signer::calls`], [`Signer::rpc`].
 #[derive(Clone)]
 pub struct Signer {
+    signer: Arc<SignerInner>,
```

Comment:
```rust
pub struct Signer {
  inner: Arc<RefCell<Inner>>
  // ...
}
```

Reply:
But how do I implement `Deref<SignerInner>` for Signer?

Reply:
```rust
impl Deref for Signer {
  target = Inner;

  fn deref(&self) -> &Inner {
    self.inner.borrow()
  }
}
```

Reply:
try `self.inner.clone().borrow()`, I remember there is a solution for it, it doesn't matter we cloning it since we are using smart pointer

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=0185a54060d277dda11ff8d85ee5e5d3

Reply:
we can re-export these APIs by introducing new methods momentary, keep all of the current tests working, for the solution of `Deref`, we create a new issue for it

Reply:
I think this approach is good enough because 5x Arc clones is a little overhead (and occurs not so often) and reference integrity is provided by this https://github.com/gear-tech/gear/blob/5b63e54c421cd71725a818b3e95c4616df3c4d22/gsdk/src/signer/mod.rs#L99

Reply:
hmmm, makes sense tbh, however we can leave some docs on it for the future refactoring is necessary

Reply:
You mean docs on line 99?

Reply:
yeah 

- `L42`: sth like `NOTE: figure a better layout for this struct in the future`
- `L99`: sth like `TODO: split calls into pallets`